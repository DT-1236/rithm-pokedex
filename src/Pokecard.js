import React, { Component } from 'react';

class Pokecard extends Component {
  render() {
    const imgUrl = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${
      this.props.id
    }.png`;

    return (
      <div className="card col-3">
        <h3>{this.props.name}</h3>
        <img src={imgUrl} alt="" className="align-self-center" />
        <p>Type: {this.props.type}</p>
        <p>EXP: {this.props.experience}</p>
      </div>
    );
  }
}

export default Pokecard;
