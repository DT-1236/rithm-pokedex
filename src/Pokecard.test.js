import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Pokecard from './Pokecard';

// smoke test
it('renders without crashing', () => {
  shallow(<Pokecard />);
});

// snapshot test
it('matches snapshot', () => {
  let wrapper = shallow(<Pokecard />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
