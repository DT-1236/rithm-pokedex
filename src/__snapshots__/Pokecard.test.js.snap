// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card"
>
  <h3 />
  <img
    alt=""
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
  />
  <p>
    Type: 
  </p>
  <p>
    EXP: 
  </p>
</div>
`;
